core/theme.js

```diff
@@ -13,65 +13,60 @@ goog.provide('Blockly.Theme');
 
 goog.require('Blockly.utils');
 goog.require('Blockly.utils.colour');
+goog.require('Blockly.utils.object');
 
 
 /**
  * Class for a theme.
  * @param {string} name Theme name.
- * @param {!Object.<string, Blockly.Theme.BlockStyle>} blockStyles A map from
- *     style names (strings) to objects with style attributes for blocks.
- * @param {!Object.<string, Blockly.Theme.CategoryStyle>} categoryStyles A map
- *     from style names (strings) to objects with style attributes for
+ * @param {!Object.<string, Blockly.Theme.BlockStyle>=} opt_blockStyles A map
+ *     from style names (strings) to objects with style attributes for blocks.
+ * @param {!Object.<string, Blockly.Theme.CategoryStyle>=} opt_categoryStyles A
+ *     map from style names (strings) to objects with style attributes for
  *     categories.
  * @param {!Object.<string, *>=} opt_componentStyles A map of Blockly component
  *     names to style value.
  * @constructor
  */
-Blockly.Theme = function(name, blockStyles, categoryStyles,
+Blockly.Theme = function(name, opt_blockStyles, opt_categoryStyles,
     opt_componentStyles) {
 
   /**
    * The theme name. This can be used to reference a specific theme in CSS.
    * @type {string}
-   * @package
    */
   this.name = name;
 
   /**
    * The block styles map.
    * @type {!Object.<string, !Blockly.Theme.BlockStyle>}
-   * @package
    */
-  this.blockStyles = blockStyles;
+  this.blockStyles = opt_blockStyles || Object.create(null);
 
   /**
    * The category styles map.
    * @type {!Object.<string, Blockly.Theme.CategoryStyle>}
-   * @package
    */
-  this.categoryStyles = categoryStyles;
+  this.categoryStyles = opt_categoryStyles || Object.create(null);
 
   /**
    * The UI components styles map.
    * @type {!Object.<string, *>}
-   * @private
    */
-  this.componentStyles_ = opt_componentStyles || Object.create(null);
+  this.componentStyles = opt_componentStyles || Object.create(null);
```

Comment:
Why are we making this public?

Reply:
I guess it could technically remain package, but certainly not protected because its being accessed from Blockly.Theme.defineTheme which is a static class method.

Reply:
Done.